aioredlock/redis.py

```diff
@@ -137,8 +163,9 @@ async def set_lock(self, resource, lock_identifier, lock_timeout):
 
         try:
             with await self.connect() as redis:
-                await redis.eval(
-                    self.set_lock_script,
+                await self._check_scripts(redis)
```

Comment:
And can we first call the script, catch the error, upload the script, and retry again?
This will halve the number of round trips.

Reply:
Ahh yes, good point.